File: Python/Problem 4.py
def is_palin(arg):
    arg = str(arg)
    for i in range(0, len(arg) // 2, 1):
        if(arg[i] != arg[-1 - i]):
            return False

    return True


temp = 0
biggest = 0
for i in range(100,1000,1):
    for j in range(100,1000,1):
        temp = i*j
        if (is_palin(temp) and temp > biggest):
            biggest = temp

print(biggest)

File: C++/Problem 4.cpp
#include <iostream>
#include <string>
using namespace std;

bool is_palin(unsigned int);

int main() {
	unsigned int temp;
	unsigned int biggest = 0;
	for(unsigned int i = 100; i < 1000; i++){
		for(unsigned int j = 100; j < 1000; j++){
			temp = i*j;
			if(is_palin(temp) & (temp > biggest)){
				biggest = temp;
			}
		}
	}

	cout << biggest << '\n';
	
	return 0;
}

bool is_palin(unsigned int arg){
	string n = to_string(arg);
	for(unsigned int i = 0; i < n.length(); i++){
		if( n[i] != n[n.length() - i - 1] ){
			return false;
		}
	}
	return true;
}
